formulas/functions/reference.js

```diff
@@ -372,6 +406,200 @@ const ReferenceFunctions = {
       return tableArray[index][colIndexNum - 1];
     }
   },
+  /***
+   * @param lookup_value: the value we are looking for in our lookup_array
+   * @param lookup_array: the array were we search for lookup_value
+   * @param return_array: the array where our return value is
+   * @param if_not_found: OPTIONAL default value if lookup_value is not found
+   * @param match_mode: OPTIONAL:
+   *                         0: If none found, return #N/A. This is the default
+   *                         -1 - Exact match. If none found, return the next smaller item.
+   *                          1 - Exact match. If none found, return the next larger item.
+   *                          2 - A wildcard match where *, ?, and ~ have special meaning.
+   * @param search_mode: OPTIONAL:
+   *                          1 - Perform a search starting at the first item. This is the default.
+   *                         -1 - Perform a reverse search starting at the last item.
+   *                          2 - Perform a binary search that relies on lookup_array being sorted in ascending order. If not sorted, invalid results will be returned.
+   *                         -2 - Perform a binary search that relies on lookup_array being sorted in descending order. If not sorted, invalid results will be returned.
+   * Microsoft Link: https://support.microsoft.com/en-us/office/xlookup-function-b7fd680e-6d10-43e6-84f9-88eae8bf5929
+   ***/
+  XLOOKUP: (
```

Comment:
This can't be in this PR

Reply:
A side effect of my squashes, I have been trying to fix it